src/antsibull/collection_meta.py

```diff
@@ -0,0 +1,54 @@
+# Author: Felix Fontein <felix@fontein.de>
+# Author: Toshio Kuratomi <tkuratom@redhat.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Ansible Project, 2020
+
+"""
+Classes to encapsulate collection metadata from collection-meta.yaml
+"""
+import typing as t
+import os
+
+from antsibull_core.yaml import load_yaml_file
+
+
+class CollectionMetadata:
+    '''
+    Stores metadata about one collection.
+    '''
+
+    changelog_url: t.Optional[str]
```

Comment:
```suggestion
    changelog_url: t.Optional[str]
    collection_directory: t.Optional[str]
    repository: t.Optional[str]
```

Reply:
Then you have to read through code when you want to know the types and attributes the class has. I prefer the attributes declared separately.